### ISSUE TITLE
"Temporary failure in name resolution" when running aarch64 binfmt images

### ISSUE THREAD

**Issue by bassam** [docker/for-mac#1621]
### Expected behavior

On docker for mac I expect the following to succeed:

```
docker run -it aarch64/ubuntu bash -c "apt-get update && apt-get install -yy -q --no-install-recommends wget && wget www.google.com"
```

### Actual behavior

On Version 17.03.1-ce-mac12 (17661):

```
> docker run -it aarch64/ubuntu bash -c "apt-get update && apt-get install -yy -q --no-install-recommends wget && wget www.google.com"
Err:1 http://ports.ubuntu.com/ubuntu-ports xenial InRelease
  Temporary failure resolving 'ports.ubuntu.com'
Err:2 http://ports.ubuntu.com/ubuntu-ports xenial-updates InRelease
  Temporary failure resolving 'ports.ubuntu.com'
Err:3 http://ports.ubuntu.com/ubuntu-ports xenial-security InRelease
  Temporary failure resolving 'ports.ubuntu.com'
Reading package lists... Done
W: Failed to fetch http://ports.ubuntu.com/ubuntu-ports/dists/xenial/InRelease  Temporary failure resolving 'ports.ubuntu.com'
W: Failed to fetch http://ports.ubuntu.com/ubuntu-ports/dists/xenial-updates/InRelease  Temporary failure resolving 'ports.ubuntu.com'
W: Failed to fetch http://ports.ubuntu.com/ubuntu-ports/dists/xenial-security/InRelease  Temporary failure resolving 'ports.ubuntu.com'
W: Some index files failed to download. They have been ignored, or old ones used instead.
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package wget
```

On Version 17.05.0-ce-mac11 (17656):

```
> docker run -it aarch64/ubuntu bash -c "apt-get update && apt-get install -yy -q --no-install-recommends wget && wget www.google.com"
Unable to find image 'aarch64/ubuntu:latest' locally
latest: Pulling from aarch64/ubuntu
8944e166ce98: Pull complete 
1f8c28d8b7a9: Pull complete 
9dbf9f348171: Pull complete 
5cb9db57901b: Pull complete 
7e21759f0cbf: Pull complete 
Digest: sha256:e76f340967a53686298c57e4d388218652f01dc9b4905e8bd32b9d2bcbc4cfa4
Status: Downloaded newer image for aarch64/ubuntu:latest
Get:1 http://ports.ubuntu.com/ubuntu-ports xenial InRelease [247 kB]
Get:2 http://ports.ubuntu.com/ubuntu-ports xenial-updates InRelease [102 kB]
Get:3 http://ports.ubuntu.com/ubuntu-ports xenial-security InRelease [102 kB]
Get:4 http://ports.ubuntu.com/ubuntu-ports xenial/main Sources [1103 kB]
Get:5 http://ports.ubuntu.com/ubuntu-ports xenial/restricted Sources [5179 B]
Get:6 http://ports.ubuntu.com/ubuntu-ports xenial/universe Sources [9802 kB]
Get:7 http://ports.ubuntu.com/ubuntu-ports xenial/main arm64 Packages [1466 kB]                                                                                                    
Get:8 http://ports.ubuntu.com/ubuntu-ports xenial/universe arm64 Packages [9493 kB]                                                                                                
Get:9 http://ports.ubuntu.com/ubuntu-ports xenial-updates/main Sources [311 kB]                                                                                                    
Get:10 http://ports.ubuntu.com/ubuntu-ports xenial-updates/restricted Sources [3202 B]                                                                                             
Get:11 http://ports.ubuntu.com/ubuntu-ports xenial-updates/universe Sources [191 kB]                                                                                               
Get:12 http://ports.ubuntu.com/ubuntu-ports xenial-updates/main arm64 Packages [596 kB]                                                                                            
Get:13 http://ports.ubuntu.com/ubuntu-ports xenial-updates/universe arm64 Packages [539 kB]                                                                                        
Get:14 http://ports.ubuntu.com/ubuntu-ports xenial-security/main Sources [84.9 kB]                                                                                                 
Get:15 http://ports.ubuntu.com/ubuntu-ports xenial-security/restricted Sources [2779 B]                                                                                            
Get:16 http://ports.ubuntu.com/ubuntu-ports xenial-security/universe Sources [31.3 kB]                                                                                             
Get:17 http://ports.ubuntu.com/ubuntu-ports xenial-security/main arm64 Packages [262 kB]                                                                                           
Get:18 http://ports.ubuntu.com/ubuntu-ports xenial-security/universe arm64 Packages [136 kB]                                                                                       
Fetched 24.5 MB in 18s (1342 kB/s)                                                                                                                                                 
Reading package lists... Done
Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  libidn11 libssl1.0.0
Recommended packages:
  ca-certificates
The following NEW packages will be installed:
  libidn11 libssl1.0.0 wget
0 upgraded, 3 newly installed, 0 to remove and 27 not upgraded.
Need to get 1008 kB of archives.
After this operation, 3787 kB of additional disk space will be used.
Get:1 http://ports.ubuntu.com/ubuntu-ports xenial-updates/main arm64 libidn11 arm64 1.32-3ubuntu1.1 [42.9 kB]
Get:2 http://ports.ubuntu.com/ubuntu-ports xenial-updates/main arm64 libssl1.0.0 arm64 1.0.2g-1ubuntu4.6 [692 kB]
Get:3 http://ports.ubuntu.com/ubuntu-ports xenial-updates/main arm64 wget arm64 1.17.1-1ubuntu1.2 [273 kB]
Fetched 1008 kB in 1s (622 kB/s)
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package libidn11:arm64.
(Reading database ... 7245 files and directories currently installed.)
Preparing to unpack .../libidn11_1.32-3ubuntu1.1_arm64.deb ...
Unpacking libidn11:arm64 (1.32-3ubuntu1.1) ...
Selecting previously unselected package libssl1.0.0:arm64.
Preparing to unpack .../libssl1.0.0_1.0.2g-1ubuntu4.6_arm64.deb ...
Unpacking libssl1.0.0:arm64 (1.0.2g-1ubuntu4.6) ...
Selecting previously unselected package wget.
Preparing to unpack .../wget_1.17.1-1ubuntu1.2_arm64.deb ...
Unpacking wget (1.17.1-1ubuntu1.2) ...
Processing triggers for libc-bin (2.23-0ubuntu5) ...
Setting up libidn11:arm64 (1.32-3ubuntu1.1) ...
Setting up libssl1.0.0:arm64 (1.0.2g-1ubuntu4.6) ...
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 76.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (Can't locate Term/ReadLine.pm in @INC (you may need to install the Term::ReadLine module) (@INC contains: /etc/perl /usr/local/lib/aarch64-linux-gnu/perl/5.22.1 /usr/local/share/perl/5.22.1 /usr/lib/aarch64-linux-gnu/perl5/5.22 /usr/share/perl5 /usr/lib/aarch64-linux-gnu/perl/5.22 /usr/share/perl/5.22 /usr/local/lib/site_perl /usr/lib/aarch64-linux-gnu/perl-base .) at /usr/share/perl5/Debconf/FrontEnd/Readline.pm line 7.)
debconf: falling back to frontend: Teletype
Setting up wget (1.17.1-1ubuntu1.2) ...
Processing triggers for libc-bin (2.23-0ubuntu5) ...
--2017-05-16 22:25:18--  http://www.google.com/
Resolving www.google.com (www.google.com)... failed: Temporary failure in name resolution.
wget: unable to resolve host address 'www.google.com'
```

### Notes

- this only happens for aarch64  (armhf seems to work ok)
- this does not happen on my ubuntu box with 4.10 kernel and docker 1.12

**Comment by likan999** [comment]
I am hitting the same problem on 18.03.1-ce-mac65. Not sure where the problem is.

**Comment by ashwoods** [comment]
+1, hit this nice bug today. updating qemu static libraries to 2.12 didn't help.

**Comment by bassam** [comment]
this is looking like a regression in recent builds of docker for Mac. Can one of the maintainers please help verify this? We're seeing this too with `18.03.1-ce-mac65`. Simple repro:

```
docker run -it  ceph/daemon-base:v3.0.5-stable-3.0-luminous-centos-7-aarch64
yum install net-tools
```

**Comment by bassam** [comment]
Looks like this was fixed in `Version 18.06.1-ce-mac73 (26764)` or prior. Closing.
